### PR TITLE
Streamline conditional compilation symbols

### LINKED ISSUE #2484: Proposal to streamline use of conditional compilation symbols
With the recent removal of NETSTANDARD support, these symbols in use can be simplified / rephrased:

!NETFRAMEWORK => NET6_0_OR_GREATER
!NET50_OR_LATER => NETFRAMEWORK
!NET6_0_OR_GREATER => NETFRAMEWORK
NET7_0_OR_GREATER => NET8_0_OR_GREATER
!NET7_0_OR_GREATER => !NET8_0_OR_GREATER
NETCOREAPP => NET6_0_OR_GREATER
net462 => NETFRAMEWORK

This will result in the following symbols being present in the code base:

NETFRAMEWORK
NET6_0_OR_GREATER
NET8_0_OR_GREATER

!NET8_0_OR_GREATER (can be removed later, when .NET 6 support is dropped)
NET6_0 (can be removed later, when .NET 6 support is dropped)

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives#conditional-compilation

### PR DESCRIPTION
**Pull request by ErikEJ**
fixes #2484

### CODE REVIEW

**Comment by ErikEJ** [comment]
@DavoudEshtehari @JRahnama FYI

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SSPI/ManagedSSPIContextProvider.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿#if !NETFRAMEWORK && !NET7_0_OR_GREATER
+﻿#if NET6_0
```
Comment (ErikEJ) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SSPI/ManagedSSPIContextProvider.cs:1]: Could also leave this as is ?

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/ValueUtilsSmi.cs [hunk 2]
```diff
@@ -3174,7 +3174,7 @@ private static SqlMoney GetSqlMoney_Unchecked(SmiEventSink_Default sink, ITypedG
 
             long temp = getters.GetInt64(sink, ordinal);
             sink.ProcessMessagesAndThrow();
-#if NETCOREAPP && NET7_0_OR_GREATER
+#if NET7_0_OR_GREATER
```
Comment (ErikEJ) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/ValueUtilsSmi.cs:3177]: Wonder if NET7_0_OR_GREATER could be replaced with NET8_0_OR_GREATER now?

Comment (ErikEJ) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/ValueUtilsSmi.cs:3177]: @roji Maybe you have some input?

Comment (roji) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/ValueUtilsSmi.cs:3177]: Yep, makes sense to me.

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SSPI/ManagedSSPIContextProvider.cs (hunk 1 above)
Comment (ErikEJ) [reply src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SSPI/ManagedSSPIContextProvider.cs:1]: Agree!

**Review by David-Engel — APPROVED** [review]

**Review by JRahnama — APPROVED** [review]

**Pull request merged**